clang/lib/Basic/Targets/RISCV.cpp

```diff
@@ -391,7 +391,14 @@ void RISCVTargetInfo::fillValidTuneCPUList(
 
 static void handleFullArchString(StringRef FullArchStr,
                                  std::vector<std::string> &Features) {
-  Features.push_back("__RISCV_TargetAttrNeedOverride");
+
+  // Should be full arch string.
+  if (!FullArchStr.starts_with("rv")) {
+    Features.push_back(FullArchStr.str());
```

Comment:
I am little confused about this part, comment say it should be full arch string, but the check come with `!`?

Reply:
Sorry for the imprecise comment. The behavior is as follows: when the input FullArchStr is invalid, it is simply passed to the next stage. The `checkTargetAttr` function will handle the invalid arch string and emit the corresponding warning.

For example, `int __attribute__((target("arch=zba")))`.

I think the comment could be rewrite as `reject the invalid FullArchStr`.

Reply:
Oh, I found the `parseArchString` also check whether FullArchStr start with "rv". 

I removed this statement and drop the unnecessary "+" sign to trigger the Sema check.